### ISSUE TITLE
Destroy cluster "You may only modify the groupSet attribute for VPC instances"

### ISSUE THREAD

**Issue by dhulse** [nchammas/flintrock#85]
When I run `flintrock destroy <cluster>` at a cluster outside of the VPC in AWS I get the message "You may only modify the groupSet attribute for VPC instances". I have noticed this is an issue for flintrock versions 0.2.0 and 0.3.0, but not for 0.1.0.

This is what I have for my config
```yaml
    # vpc-id: <id>
    # subnet-id: <id>
```

Here is the full output
```bash
flintrock destroy dale-lsh
dale-lsh:
  state: running
  node-count: 201
  master: ec2-50-19-147-60.compute-1.amazonaws.com
  slaves:
    - ec2-23-22-140-251.compute-1.amazonaws.com
    - ec2-50-17-126-74.compute-1.amazonaws.com
    - ec2-107-21-137-51.compute-1.amazonaws.com
    - ec2-54-91-95-67.compute-1.amazonaws.com
    - ec2-54-91-153-113.compute-1.amazonaws.com
    - ec2-54-227-87-108.compute-1.amazonaws.com
    - ec2-54-82-66-138.compute-1.amazonaws.com
    - ec2-54-221-0-134.compute-1.amazonaws.com
    - ec2-50-17-83-252.compute-1.amazonaws.com
    - ec2-54-167-26-223.compute-1.amazonaws.com
    - ec2-54-92-192-178.compute-1.amazonaws.com
    - ec2-54-90-150-249.compute-1.amazonaws.com
    - ec2-54-242-129-235.compute-1.amazonaws.com
    - ec2-54-87-28-208.compute-1.amazonaws.com
    - ec2-54-159-39-90.compute-1.amazonaws.com
    - ec2-54-242-160-161.compute-1.amazonaws.com
    - ec2-54-196-177-62.compute-1.amazonaws.com
    - ec2-54-144-65-225.compute-1.amazonaws.com
    - ec2-54-82-84-171.compute-1.amazonaws.com
    - ec2-54-227-60-71.compute-1.amazonaws.com
    - ec2-54-82-7-228.compute-1.amazonaws.com
    - ec2-54-221-111-109.compute-1.amazonaws.com
    - ec2-54-196-217-0.compute-1.amazonaws.com
    - ec2-54-226-248-200.compute-1.amazonaws.com
    - ec2-50-17-122-243.compute-1.amazonaws.com
    - ec2-54-160-203-126.compute-1.amazonaws.com
    - ec2-54-227-9-58.compute-1.amazonaws.com
    - ec2-54-145-62-121.compute-1.amazonaws.com
    - ec2-107-21-140-165.compute-1.amazonaws.com
    - ec2-54-147-131-121.compute-1.amazonaws.com
    - ec2-54-163-39-161.compute-1.amazonaws.com
    - ec2-54-147-168-163.compute-1.amazonaws.com
    - ec2-23-22-176-58.compute-1.amazonaws.com
    - ec2-54-147-182-91.compute-1.amazonaws.com
    - ec2-107-21-182-183.compute-1.amazonaws.com
    - ec2-54-147-11-170.compute-1.amazonaws.com
    - ec2-54-196-252-58.compute-1.amazonaws.com
    - ec2-54-90-180-190.compute-1.amazonaws.com
    - ec2-54-226-53-187.compute-1.amazonaws.com
    - ec2-54-167-98-196.compute-1.amazonaws.com
    - ec2-54-242-101-14.compute-1.amazonaws.com
    - ec2-54-90-163-36.compute-1.amazonaws.com
    - ec2-54-144-87-126.compute-1.amazonaws.com
    - ec2-54-82-81-212.compute-1.amazonaws.com
    - ec2-54-83-77-11.compute-1.amazonaws.com
    - ec2-54-221-98-203.compute-1.amazonaws.com
    - ec2-54-242-125-165.compute-1.amazonaws.com
    - ec2-54-221-150-81.compute-1.amazonaws.com
    - ec2-54-83-91-99.compute-1.amazonaws.com
    - ec2-54-160-134-176.compute-1.amazonaws.com
    - ec2-54-83-111-115.compute-1.amazonaws.com
    - ec2-54-92-217-202.compute-1.amazonaws.com
    - ec2-54-146-201-97.compute-1.amazonaws.com
    - ec2-23-22-170-108.compute-1.amazonaws.com
    - ec2-54-196-164-225.compute-1.amazonaws.com
    - ec2-54-221-12-243.compute-1.amazonaws.com
    - ec2-54-145-30-139.compute-1.amazonaws.com
    - ec2-54-234-31-227.compute-1.amazonaws.com
    - ec2-54-242-3-71.compute-1.amazonaws.com
    - ec2-54-167-33-215.compute-1.amazonaws.com
    - ec2-54-221-130-212.compute-1.amazonaws.com
    - ec2-54-160-193-240.compute-1.amazonaws.com
    - ec2-54-242-85-70.compute-1.amazonaws.com
    - ec2-54-196-252-93.compute-1.amazonaws.com
    - ec2-54-82-16-116.compute-1.amazonaws.com
    - ec2-54-226-5-51.compute-1.amazonaws.com
    - ec2-54-82-93-114.compute-1.amazonaws.com
    - ec2-54-196-217-119.compute-1.amazonaws.com
    - ec2-54-146-253-138.compute-1.amazonaws.com
    - ec2-54-234-169-195.compute-1.amazonaws.com
    - ec2-54-147-4-167.compute-1.amazonaws.com
    - ec2-54-163-20-244.compute-1.amazonaws.com
    - ec2-54-91-151-170.compute-1.amazonaws.com
    - ec2-54-160-170-38.compute-1.amazonaws.com
    - ec2-50-19-24-139.compute-1.amazonaws.com
    - ec2-54-242-39-79.compute-1.amazonaws.com
    - ec2-54-196-252-148.compute-1.amazonaws.com
    - ec2-54-90-131-236.compute-1.amazonaws.com
    - ec2-54-145-23-5.compute-1.amazonaws.com
    - ec2-54-90-141-112.compute-1.amazonaws.com
    - ec2-54-91-161-130.compute-1.amazonaws.com
    - ec2-54-234-205-204.compute-1.amazonaws.com
    - ec2-54-234-174-232.compute-1.amazonaws.com
    - ec2-54-221-52-6.compute-1.amazonaws.com
    - ec2-54-90-222-69.compute-1.amazonaws.com
    - ec2-50-17-96-127.compute-1.amazonaws.com
    - ec2-54-234-74-87.compute-1.amazonaws.com
    - ec2-54-145-48-130.compute-1.amazonaws.com
    - ec2-54-87-19-80.compute-1.amazonaws.com
    - ec2-54-145-29-229.compute-1.amazonaws.com
    - ec2-54-167-81-203.compute-1.amazonaws.com
    - ec2-54-163-201-254.compute-1.amazonaws.com
    - ec2-54-234-39-250.compute-1.amazonaws.com
    - ec2-54-227-43-155.compute-1.amazonaws.com
    - ec2-54-234-153-24.compute-1.amazonaws.com
    - ec2-107-20-99-232.compute-1.amazonaws.com
    - ec2-54-226-248-225.compute-1.amazonaws.com
    - ec2-54-242-164-178.compute-1.amazonaws.com
    - ec2-50-19-55-113.compute-1.amazonaws.com
    - ec2-54-145-49-16.compute-1.amazonaws.com
    - ec2-54-90-164-138.compute-1.amazonaws.com
    - ec2-54-242-113-241.compute-1.amazonaws.com
    - ec2-54-234-242-55.compute-1.amazonaws.com
    - ec2-54-196-219-173.compute-1.amazonaws.com
    - ec2-54-146-225-132.compute-1.amazonaws.com
    - ec2-54-167-98-222.compute-1.amazonaws.com
    - ec2-23-22-235-210.compute-1.amazonaws.com
    - ec2-54-196-155-223.compute-1.amazonaws.com
    - ec2-54-227-63-240.compute-1.amazonaws.com
    - ec2-54-163-221-149.compute-1.amazonaws.com
    - ec2-23-22-128-131.compute-1.amazonaws.com
    - ec2-54-90-140-98.compute-1.amazonaws.com
    - ec2-54-146-225-214.compute-1.amazonaws.com
    - ec2-54-227-99-115.compute-1.amazonaws.com
    - ec2-54-226-72-96.compute-1.amazonaws.com
    - ec2-107-21-179-244.compute-1.amazonaws.com
    - ec2-54-196-154-183.compute-1.amazonaws.com
    - ec2-54-242-150-234.compute-1.amazonaws.com
    - ec2-54-234-82-196.compute-1.amazonaws.com
    - ec2-107-21-149-201.compute-1.amazonaws.com
    - ec2-54-242-137-4.compute-1.amazonaws.com
    - ec2-54-221-3-134.compute-1.amazonaws.com
    - ec2-54-87-50-255.compute-1.amazonaws.com
    - ec2-54-83-97-25.compute-1.amazonaws.com
    - ec2-54-82-41-132.compute-1.amazonaws.com
    - ec2-54-242-144-253.compute-1.amazonaws.com
    - ec2-107-20-75-192.compute-1.amazonaws.com
    - ec2-54-90-64-117.compute-1.amazonaws.com
    - ec2-54-227-74-149.compute-1.amazonaws.com
    - ec2-54-221-106-93.compute-1.amazonaws.com
    - ec2-54-242-155-101.compute-1.amazonaws.com
    - ec2-107-20-56-139.compute-1.amazonaws.com
    - ec2-54-159-21-137.compute-1.amazonaws.com
    - ec2-54-226-98-59.compute-1.amazonaws.com
    - ec2-54-167-72-19.compute-1.amazonaws.com
    - ec2-54-91-133-83.compute-1.amazonaws.com
    - ec2-54-87-27-214.compute-1.amazonaws.com
    - ec2-23-22-155-98.compute-1.amazonaws.com
    - ec2-54-83-126-252.compute-1.amazonaws.com
    - ec2-54-234-193-71.compute-1.amazonaws.com
    - ec2-54-226-237-110.compute-1.amazonaws.com
    - ec2-54-144-98-168.compute-1.amazonaws.com
    - ec2-50-17-102-207.compute-1.amazonaws.com
    - ec2-54-196-251-245.compute-1.amazonaws.com
    - ec2-54-160-253-189.compute-1.amazonaws.com
    - ec2-50-17-107-48.compute-1.amazonaws.com
    - ec2-54-144-82-7.compute-1.amazonaws.com
    - ec2-54-196-220-111.compute-1.amazonaws.com
    - ec2-54-167-55-184.compute-1.amazonaws.com
    - ec2-54-160-163-166.compute-1.amazonaws.com
    - ec2-23-22-178-67.compute-1.amazonaws.com
    - ec2-107-20-18-109.compute-1.amazonaws.com
    - ec2-54-196-180-60.compute-1.amazonaws.com
    - ec2-54-82-80-169.compute-1.amazonaws.com
    - ec2-54-167-125-16.compute-1.amazonaws.com
    - ec2-54-234-100-79.compute-1.amazonaws.com
    - ec2-54-90-70-2.compute-1.amazonaws.com
    - ec2-107-20-97-152.compute-1.amazonaws.com
    - ec2-107-20-68-197.compute-1.amazonaws.com
    - ec2-54-146-205-66.compute-1.amazonaws.com
    - ec2-54-147-154-109.compute-1.amazonaws.com
    - ec2-54-160-145-147.compute-1.amazonaws.com
    - ec2-54-242-113-71.compute-1.amazonaws.com
    - ec2-54-91-104-208.compute-1.amazonaws.com
    - ec2-54-90-184-255.compute-1.amazonaws.com
    - ec2-54-226-177-118.compute-1.amazonaws.com
    - ec2-54-221-29-172.compute-1.amazonaws.com
    - ec2-54-196-157-69.compute-1.amazonaws.com
    - ec2-54-242-32-246.compute-1.amazonaws.com
    - ec2-54-242-8-111.compute-1.amazonaws.com
    - ec2-54-91-113-225.compute-1.amazonaws.com
    - ec2-54-234-235-212.compute-1.amazonaws.com
    - ec2-54-242-42-94.compute-1.amazonaws.com
    - ec2-50-17-72-112.compute-1.amazonaws.com
    - ec2-107-20-14-218.compute-1.amazonaws.com
    - ec2-54-163-59-155.compute-1.amazonaws.com
    - ec2-184-72-140-54.compute-1.amazonaws.com
    - ec2-23-22-164-42.compute-1.amazonaws.com
    - ec2-54-234-124-12.compute-1.amazonaws.com
    - ec2-54-163-40-117.compute-1.amazonaws.com
    - ec2-54-163-194-209.compute-1.amazonaws.com
    - ec2-54-144-95-31.compute-1.amazonaws.com
    - ec2-54-144-85-65.compute-1.amazonaws.com
    - ec2-54-91-203-191.compute-1.amazonaws.com
    - ec2-54-87-32-41.compute-1.amazonaws.com
    - ec2-54-90-230-217.compute-1.amazonaws.com
    - ec2-54-145-17-101.compute-1.amazonaws.com
    - ec2-54-196-252-146.compute-1.amazonaws.com
    - ec2-107-20-93-138.compute-1.amazonaws.com
    - ec2-23-22-169-33.compute-1.amazonaws.com
    - ec2-107-20-12-25.compute-1.amazonaws.com
    - ec2-54-90-138-73.compute-1.amazonaws.com
    - ec2-23-22-210-97.compute-1.amazonaws.com
    - ec2-54-159-14-101.compute-1.amazonaws.com
    - ec2-54-92-195-82.compute-1.amazonaws.com
    - ec2-54-159-0-151.compute-1.amazonaws.com
    - ec2-50-17-84-80.compute-1.amazonaws.com
    - ec2-54-221-6-237.compute-1.amazonaws.com
    - ec2-54-147-143-89.compute-1.amazonaws.com
    - ec2-54-163-223-132.compute-1.amazonaws.com
Are you sure you want to destroy this cluster? [y/N]: y
Destroying dale-lsh...
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/flintrock.py", line 804, in main
    cli(obj={})
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/flintrock.py", line 332, in destroy
    cluster.destroy()
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/ec2.py", line 118, in destroy
    Groups=[flintrock_base_group.id])
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/boto3/resources/factory.py", line 455, in do_action
    response = action(self, *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/boto3/resources/action.py", line 79, in __call__
    response = getattr(parent.meta.client, operation_name)(**params)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/botocore/client.py", line 310, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/botocore/client.py", line 407, in _make_api_call
    raise ClientError(parsed_response, operation_name)
An error occurred (InvalidParameterCombination) when calling the ModifyInstanceAttribute operation: You may only modify the groupSet attribute for VPC instances
```

**Comment by nchammas** [comment]
The root issue appears to be with [this call](https://github.com/nchammas/flintrock/blob/e163743fcc4d2187e786020c91052e67ece48c74/flintrock/ec2.py#L117-L118), where I "unassign" the cluster security group from the instance. [`modify_attribute()`](https://boto3.readthedocs.org/en/latest/reference/services/ec2.html#EC2.Instance.modify_attribute) only works with instances in a VPC.

I'll have to think about how to get around this. The reason I "unassign" the security groups there is, well, here's [what I wrote](https://github.com/nchammas/flintrock/blob/e163743fcc4d2187e786020c91052e67ece48c74/flintrock/ec2.py#L110-L113):

>        # We "unassign" the cluster security group here (i.e. the
>        # 'flintrock-clustername' group) so that we can immediately delete it once
>        # the instances are terminated. If we don't do this, we get dependency
>        # violations for a couple of minutes before we can actually delete the group.

If there are no other options, maybe we have to eat the 1-2 minute wait between when we destroy the instances and when Amazon will let us delete the cluster security group. :/ That will work with both VPC and non-VPC clusters.

Another approach is to refuse to launch clusters outside of a VPC.

I find this more attractive since it makes Flintrock simpler, and also because Amazon seems to be going in the direction of [discouraging use of EC2 Classic](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-vpc.html). The most telling indication is this statement:

> If you created your account after 2013-12-04, it supports EC2-VPC only.

@dhulse - Would you be devastated if Flintrock only supported launching into VPCs?

The reason this doesn't happen in 0.1.0 is that we used to leave the cluster security group lying around after the cluster got terminated, which was essentially clutter.

**Comment by dhulse** [comment]
I understand why you only want to support VPC clusters. I am currently setting things up to launch a cluster in a VPC using flintrock. When I try to launch a cluster into a VPC I get this error "Exception: Error authorizing cluster ingress to self."  I am not sure if the error is a bug in flintrock or if there is something else I need configured for my AWS VPC.

Here is the output when I attempt to launch a cluster into a VPC:
```bash
flintrock launch test-vpc --num-slaves 2
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/ec2.py", line 323, in get_or_create_ec2_security_groups
    SourceSecurityGroupName=cluster_group.group_name)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/boto3/resources/factory.py", line 455, in do_action
    response = action(self, *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/boto3/resources/action.py", line 79, in __call__
    response = getattr(parent.meta.client, operation_name)(**params)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/botocore/client.py", line 310, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/botocore/client.py", line 407, in _make_api_call
    raise ClientError(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (InvalidParameterValue) when calling the AuthorizeSecurityGroupIngress operation: Invalid value 'flintrock-test-vpc' for groupName. You may not reference Amazon VPC security groups by name. Please use the corresponding id for this operation.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.5/bin/flintrock", line 9, in <module>
    load_entry_point('Flintrock==0.4.0.dev0', 'console_scripts', 'flintrock')()
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/flintrock.py", line 803, in main
    cli(obj={})
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/flintrock.py", line 295, in launch
    instance_initiated_shutdown_behavior=ec2_instance_initiated_shutdown_behavior)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/ec2.py", line 30, in wrapper
    res = func(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/ec2.py", line 409, in launch
    region=region)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/ec2.py", line 326, in get_or_create_ec2_security_groups
    raise Exception("Error authorizing cluster ingress to self.") from e
Exception: Error authorizing cluster ingress to self.
```

**Comment by nchammas** [comment]
Oh hey, I've hit this issue randomly here and there and never got to the bottom of it because I didn't have the proper error handling implemented at the time.

But it looks like you got the gold this time:

```
botocore.exceptions.ClientError: An error occurred (InvalidParameterValue) when calling the 
AuthorizeSecurityGroupIngress operation: Invalid value 'flintrock-test-vpc' for groupName.
You may not reference Amazon VPC security groups by name. Please use the corresponding id for
this operation.
```

Could you report this in a separate issue? This should be easy to fix. Also, what happens if you just retry the launch?

So with regards to VPC vs. Classic, are you OK with VPC-only support (given that it works, of course)?

**Comment by dhulse** [comment]
If I can launch a VPC cluster, I will be totally fine with VPC-only support.

I retried the launch and I do not get the same output as the previous one posted.

Here is my output on the retried launch
```bash
flintrock launch test-vpc --num-slaves 2
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/flintrock.py", line 804, in main
    cli(obj={})
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/flintrock.py", line 296, in launch
    instance_initiated_shutdown_behavior=ec2_instance_initiated_shutdown_behavior)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/ec2.py", line 30, in wrapper
    res = func(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/ec2.py", line 405, in launch
    region=region)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/flintrock/ec2.py", line 326, in get_or_create_ec2_security_groups
    raise Exception("Error authorizing cluster ingress to self.") from e
Error authorizing cluster ingress to self.
```

**Comment by marcuscollins** [comment]
@dhulse see #90 I've seen the same issue, haven't got a complete fix yet.
